src/firefly/js/metaConvert/TableDataProductUtils.js

```diff
@@ -373,3 +379,75 @@ export function findGridTableRows(table,maxRows, plotIdRoot) {
 }
 
 
+let lastTblId;
+
+/**
+ * passing an object with tbl_id as key and activate function as value. Make a new activate function
+ * @param {Object} activateObj
+ * @param {ActivateParams} activateParams
+ * @return {function(): function(): void}
+ */
+export function makeMultiTableActivate(activateObj,activateParams) {
+    return () => {
+        const loadedAry=[];
+        const deActivateAry= Object.values(activateObj).map( ({activate}) => activate());
+        const tblIdList= Object.keys(activateObj);
+        const chartList= Object.values(activateObj).map( ({chartId}) => chartId);
+
+        tblIdList.forEach( async (tbl_id) => {
```

Comment:
I find this business with `loadedAry` and making sync `forEach` to wait for all async calls hard to read and weird tbh.

why not to use composition?

`await Promise.all(tblIdList.map(async (tbl_id) => {
    await onTableLoaded(tbl_id);
}));

if (lastTblId) dispatchActiveTableChanged(lastTblId,activateParams.tableGroupViewerId);

const id= tblIdList.join('-');
`

Reply:
`Promise.all` is a good idea. I will try it like that. I can't us await since it is not in async function but I can use `Promise.all` with a `then`.